src/server/frontend/wayland/wayland_connector.cpp

```diff
@@ -1891,62 +1910,30 @@ class WlShellSurface : public wayland::ShellSurface
         wl_resource* parent,
         uint32_t id,
         wl_resource* surface,
-        std::shared_ptr<mf::Shell> const& shell,
-        WlSeat& seat)
-        : ShellSurface(client, parent, id),
-          destroyed{std::make_shared<bool>(false)},
-          shell{shell}
+        std::shared_ptr<mf::Shell> const& shell_,
+        WlSeat& seat) :
+        ShellSurface(client, parent, id),
+        destroyed{std::make_shared<bool>(false)},
+        shell{shell_}
     {
         auto* tmp = wl_resource_get_user_data(surface);
-        auto& mir_surface = *static_cast<WlSurface*>(tmp);
-
-        auto const session = session_for_client(client);
-
-        auto params = ms::SurfaceCreationParameters()
-            .of_type(mir_window_type_freestyle)
-            .of_size(geom::Size{640, 480})
-            .with_buffer_stream(mir_surface.stream_id);
+        auto* mir_surface = static_cast<WlSurface*>(tmp);
```

Comment:
Hm, I know this is existing code, but is it actually safe?

wl_resource_get_user_data(surface) will return a void* pointing to a wayland::Surface; I don't think static_cast-ing directly to a WlSurface* is guaranteed to do the right thing?

Shouldn't it be `static_cast<WlSurface*>(static_cast<wayland::Surface*>(tmp))`?

Reply:
You're right. Fix on the way.